wordpress-mobile/WordPress-iOS
# Move blog time zone lookup into blog object

Works toward #15832 by reducing the need to create as many `BlogService` objects. Fewer `BlogService` objects will result in fewer places we need to thread the `CoreDataStack` through, which will make #15832 far simpler.

For this PR, we're moving the `timeZoneForBlog:` method into the `Blog` object, which allows us to remove several ephemeral `BlogService` instances (and as a bonus, a couple of references to `ContextManager.sharedInstance()`).

@aerych – wondering if you can take a look at this, since I see you did similar work in 3ba5cda.

@leandroalonso – wondering if you'd be able to take a look at the tests for this – the approach to using `PostBuilder.setUpInMemoryManagedObjectContext()` in the `PostBuilder` constructor means that the managed object context is destroyed when `PostBuilder` passes out of scope because `NSManagedObject` only holds a weak reference to its context. I went around and fixed this (and added debug assertions to prevent regression) in the failing tests caused by my changes – hope that's ok!!

**To test:**
- The change is covered by tests, so ensure those pass in CI
- If you look at the commits from oldest to newest, they spell out the process in small mechanical steps that should help verify that this code is functionally identical to what came before.
- Ensure the new tests look ok – they should be identical – the change just makes it so the context is owned by the test, not the `Post` object.

**PR submission checklist:**
